### ISSUE TITLE
Performance affected by unused log statements

### ISSUE THREAD

**Issue by remster** [terranodo/tegola#197]
We've been trying to use tegola for on-demand mapbox tile generation and we didn't see the performance we were after. I've pprof'ed it and i saw this:
![profile](https://user-images.githubusercontent.com/8327961/33332261-1e09427e-d464-11e7-88c0-94b97d63be44.png)
I don't know if i read it right, but it appears that 40% of tile generation cpu cycles are spent on fmt.Sprintf
composing log statements. I have then commented [this](https://github.com/terranodo/tegola/blob/master/maths/makevalid/plyg/ring.go#L622) out and saw my performance test come back in about half the time it took beforehand. 

Does my observation seem plausible?

**Comment by gdey** [comment]
Yes, that's possible. I have not looked at performance tuning, and cleanup quite yet. That is left over code from when I was trying to figure out why there were certain cases of makevalid that was not working. Thank you for looking into it. We can quickly patch this, but I'm sure there are other areas that can be cleaned up and improved.

**Comment by ARolek** [comment]
@remster Thanks for the research! This is absolutely a possible performance bottleneck and a quick one to fix. For the v0.4.0 release we focused on making geo processing work, not necessarily work fast. @gdey is working on a write up of the MakeValid algorithm and the v0.5.0 release is focusing on performance. Nonetheless, this is could be a quick win. We will do some analysis and push up a patch release if this turns out to be as straight forward as it appears.

**Comment by kalxas** [comment]
👍

**Comment by ARolek** [comment]
@remster I have confirmed that you have found some low hanging performance fruit. I have cleaned up the unused debug code and will be pushing out a patch release shortly. We still have a lot of performance improvements coming through the pipe but this a such a great win that it's worth pushing a release out now. I'm seeing upwards of 40% faster response times on polygon heavy tiles like the following one from Natural Earth v4:
 
<img width="562" alt="screen shot 2017-11-28 at 3 06 26 pm" src="https://user-images.githubusercontent.com/667850/33349820-a2ec77f4-d450-11e7-9049-7d5103589c50.png">

This is off topic, but have you tried implementing a `filecache` to help with subsequent tile requests, or do all requests need to be direct from the datastore every time?
